app/models/attachment_data.rb

```diff
@@ -130,6 +130,10 @@ def visible_edition_for(user)
     visible_attachable.is_a?(Edition) ? visible_attachable : nil
   end
 
+  def draft_attachment_for(user)
+    visible_to?(user) ? attachments.find { |attachment| attachment.attachable_type == "Edition" && attachment.attachable.draft? } : nil
```

Comment:
I think it should actually pass to the fail block. I must admit i'm not sure why it shouldn't be rendered though. I'm assuming there's a good reason though? Currently the page would blow up as it uses the edition in the view.

I'll extend the same behaviour to draft editions, but i might be worth having a dig into.

Reply:
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/42515961/178247240-3a54392c-ae9b-465b-a72c-14c945dfad63.png">

So there are 12 cases where this can be true. If we want these to be able to render then we would need to do more controller/view work to ensure the page can be rendered with other attachable types. If this is worth doing i think we should probs do it in a follow up PR though. Wdyt?

Reply:
Ah, I hadn't considered that the view depended on it being an Edition. That makes sense, since this is for previewing draft CSVs – and it's only because they're attached to an Edition that they have the "draft" workflow.

Doesn't seem like the outliers are worth worrying about, since they'll never have been preview-able anyway. 👍🏻